websites/T/Taiko Web/presence.ts

```diff
@@ -0,0 +1,114 @@
+const presence = new Presence({
+    clientId: "858246998561783828"
+  }),
+  slideShow = presence.createSlideshow(),
+  browsingStamp = Math.floor(Date.now() / 1000),
+  script = document.createElement("script");
+
+/*
+  Overriding XMLHttpRequest.open method
+  to know the current request song
+  Doesn't work if song is in cache :[
+*/
+script.innerHTML = `
+  !function(open){
+    XMLHttpRequest.prototype.open = function (method, url) {
+      if (url && url.startsWith("https://taiko.uk/taiko/songs/")) {
+        let songId = parseInt(url.substr(29));
+        let song = assets.songs.find(song => song.id === songId);
+        window.song = {};
+        window.song.title_lang = song.title_lang;
+        window.song.title = song.title;
+        window.song.category = song.category;
+      }
+      open.call(this, method, url);
+    }
+  }(XMLHttpRequest.prototype.open);`;
+
+document.body.append(script);
+
+presence.on("UpdateData", async () => {
+  const presenceData: PresenceData = {
+      largeImageKey: "taiko_logo",
+      startTimestamp: browsingStamp
+    },
+    presenceData1: PresenceData = {
+      largeImageKey: "taiko_logo",
+      smallImageKey: "taiko_logo",
+      startTimestamp: browsingStamp
+    },
```

Comment:
I don't really understand what's the point of this.

Reply:
These are two slide of the slideshow? Do you want me to convert them to array of two objects?

Reply:
Yeah.